plane_wave.ipynb

```diff
@@ -0,0 +1,643 @@
+    "The length of the tube along the $x$-axis should exhibit at least half of the wavelength, but could be longer.\n",
+    "\n",
+    "At $x=0$ there is a sound rigid wall (the choice of $0$ makes calculus much simpler).\n",
+    "This results in our considered boundary condition $v(x=0) = 0$.\n",
+    "Sound rigid walls force velocity to zero at the wall, walls are rather not moving by impinging sound waves, right?!\n",
+    "\n",
+    "\n",
+    "In Ch. 2 of https://link.springer.com/book/10.1007/978-3-662-47704-5 we find the derivation of the standing wave phenomena which we are dealing here.\n",
+    "\n",
+    "We need the density of air $\\rho_0$ in the below equations."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e116e909",
+   "metadata": {},
+   "source": [
+    "## 3.1 Solution standing wave\n",
+    "\n",
+    "The solution is\n",
+    "\n",
+    "$$p(x,t) = 2 p_A \\cos(k_x x) \\mathrm{e}^{\\mathrm{j} \\omega t}$$\n",
+    "$$v(x,t) = 2 p_A \\frac{-\\mathrm{j}}{\\rho_0 c}\\sin(k_x x) \\mathrm{e}^{\\mathrm{j} \\omega t},$$\n",
+    "\n",
+    "or written with our used $\\Re$-part convention (note: $-\\mathrm{j} = \\mathrm{e}^{-\\mathrm{j}\\frac{\\pi}{2}}$, $\\cos(\\omega t-\\frac{\\pi}{2}) = \\sin(\\omega t)$)\n",
+    "\n",
+    "$$p(x,t) = 2 p_A \\cos(k_x x) \\cos(\\omega t)$$\n",
+    "$$v(x,t) = 2 p_A \\frac{1}{\\rho_0 c}\\sin(k_x x) \\sin(\\omega t)$$\n",
+    "\n",
+    "The other boundary condition $v(x=-l) = v_A$ yields the relation between $v_A$ and $p_A$ (note: use complex formulation and $-\\sin(-x) = \\sin(x)$)\n",
+    "\n",
+    "$$v_A = 2 p_A \\frac{-\\mathrm{j}}{\\rho_0 c}\\sin(k_x (-l)) = \\frac{p_A}{\\rho_0 c} 2 \\mathrm{j} \\sin(k_x l)$$\n",
+    "\n",
+    "Below, for simplicity, we used $p_A = 1 \\mathrm{N}/\\mathrm{m}^2$ and also we normalized the plots, such that the actual imposed energy is not given in absolute values."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c3bfde44",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "n_samples_per_period = 64\n",
+    "c = 343  # m/s, speed of sound\n",
+    "wave_length = 1  # m\n",
+    "pA = 1  # N / m^2 -> see above how vA is derived from that\n",
+    "rho0 = 1.2041  # kg/m^3"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f550b37a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "kx = 2*np.pi / wave_length\n",
+    "w_c = kx\n",
+    "w = w_c * c\n",
+    "Ts = 2*np.pi / (w*n_samples_per_period)  # s, sampling period\n",
+    "x = np.linspace(-2*wave_length, 0, 2**10)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8657d599",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "N = 2**7\n",
+    "x = np.linspace(-2.1334*wave_length, 0, N, endpoint=True)\n",
+    "# boundary condition for rigid wall\n",
+    "p_cpx = 2*pA * np.cos(kx*x)\n",
+    "v_cpx = 2*pA * -1j / (rho0*c) * np.sin(kx*x)\n",
+    "\n",
```

Comment:
Would be helpful to comment that the time-harmonic term (phasor) is not included and will be multiplied in the following code.